addons/account/models/account_move.py

```diff
@@ -2765,16 +2768,11 @@ def write(self, vals):
                     posted_move._check_fiscalyear_lock_date()
                     posted_move.line_ids._check_tax_lock_date()
 
-                # Hash the moves that fit the move hash domain
-                if vals.get('is_move_sent') or vals.get('state') == 'posted':
-                    self.flush_recordset()  # Ensure that the name is correctly computed before it is used to generate the hash
-                    filter_domain = self._get_move_hash_domain(common_domain=[('inalterable_hash', '=', False)])
-                    for move in self.filtered_domain(filter_domain).sorted(lambda m: (m.date, m.ref or '', m.id)):
-                        new_number = move.journal_id.secure_sequence_id.next_by_id()
-                        res |= super(AccountMove, move).write({
-                            'secure_sequence_number': new_number,
-                            'inalterable_hash': move._get_new_hash(new_number),
-                        })
+                if vals.get('state') == 'posted':
+                    self.flush_recordset()  # Ensure that the name is correctly computed
```

Comment:
The flush is needed to compute the name correctly which is done on post. So no need to reflush when the move is sent

If we flush only at `is_move_sent` (or inside `_hash_moves`), some tests will fail because the name is not correctly computed

Reply:
Which tests?
Seems like a side effect

Reply:
https://runbot.odoo.com/runbot/build/60714051

I can fix some all of them with some flushes at specific points except for `test_inter_company_invoice_flow_sub_companies`. I'm not sure why `customer_invoice.payment_reference` should be equal to `bill.payment_reference`. But I'm wondering why flushing on is_move_sent would be better than on post ? 

```diff
diff --git a/addons/account/models/account_move.py b/addons/account/models/account_move.py
index 1edd4c6906c9..d7a6960a992b 100644
--- a/addons/account/models/account_move.py
+++ b/addons/account/models/account_move.py
@@ -1688,7 +1688,7 @@ class AccountMove(models.Model):
         other_moves.abnormal_date_warning = False
         if not draft_invoices:
             return
-        draft_invoices.flush_recordset(['invoice_date', 'date', 'amount_total', 'partner_id', 'move_type', 'company_id'])
+        draft_invoices.flush_model(['invoice_date', 'date', 'amount_total', 'partner_id', 'move_type', 'company_id', 'name'])
         today = fields.Date.context_today(self)
         self.env.cr.execute("""
             WITH previous_invoices AS (
@@ -2768,10 +2768,11 @@ class AccountMove(models.Model):
                     posted_move._check_fiscalyear_lock_date()
                     posted_move.line_ids._check_tax_lock_date()
 
-                if vals.get('state') == 'posted':
-                    self.flush_recordset()  # Ensure that the name is correctly computed
+                # if vals.get('state') == 'posted':
+                #     pass
 
                 if vals.get('is_move_sent'):
+                    self.flush_recordset()  # Ensure that the name is correctly computed
                     self._hash_moves()
 
             self._synchronize_business_models(set(vals.keys()))

diff --git a/account_accountant/tests/test_reconciliation_matching_rules.py b/account_accountant/tests/test_reconciliation_matching_rules.py
index 4b36c176db6..714d1371945 100644
--- a/account_accountant/tests/test_reconciliation_matching_rules.py
+++ b/account_accountant/tests/test_reconciliation_matching_rules.py
@@ -1177,6 +1177,7 @@ class TestReconciliationMatchingRules(AccountTestInvoicingCommon):
             })],
         })
         invoice2.action_post()
+        invoice2.flush_recordset()
         term_lines = (invoice + invoice2).line_ids.filtered(lambda x: x.display_type == 'payment_term')
 
         # Matching multiple invoices.
diff --git a/account_reports/models/account_deferred_reports.py b/account_reports/models/account_deferred_reports.py
index dc7817c28ec..c3b05531d28 100644
--- a/account_reports/models/account_deferred_reports.py
+++ b/account_reports/models/account_deferred_reports.py
@@ -384,6 +384,7 @@ class DeferredReportCustomHandler(models.AbstractModel):
             raise UserError(_("You cannot generate entries for a period that is locked."))
         options['all_entries'] = False  # We only want to create deferrals for posted moves
         report = self.env["account.report"].browse(options["report_id"])
+        self.env['account.move'].flush_model()
         self.env['account.move.line'].flush_model()
         lines = self._get_lines(report, options, filter_already_generated=True)
         deferral_entry_period = self.env['account.report']._get_dates_period(date_from, date_to, 'range', period_type='month')

´´´